js-foundations.js

```diff
@@ -16,35 +38,87 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
 // 3a. Write a function that returns every other item from the array as one string. Expected output: "streetlamp teeth nephew database"
 
 
+let randomSkips = randomNouns.filter((value, index) => {
+    return index %2 === 0
+})
 
+
+
+console.log(randomSkips.join(" "))
 // 3b. Create a function that takes in the variable and returns the array with all the words capitalized. Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
+var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 
+ let firstLetterCap = randomNouns.map((value, index) => {
+    return value.toUpperCase() 
+ })
+     
+ // I could not for the life of me figure out how to make only the first letter capitalyzed and not the whole word :(
+ 
+  console.log(firstLetterCap)
 
 // 3c. STRETCH: Create a function that returns the length of each string from the randomNouns variable. Expected output: [10, 6, 5, 10, 6, 11, 8]
 
+let lengthOfEachStr = randomNouns.map((value, index) => {
+    return value.length
+ })
 
-
+console.log(lengthOfEachStr)
 
 // -------------------------------------- Consider the variables:
 var testString1 = "learn"
+
 var testString2 = "academy"
+
 var testString3 = "sandiego"
 
 // 4a. Write a function called alphabetSoup that takes one argument and returns a string with all the letters in alphabetical order. Use each of the varibales as test cases. Expected output: "aelnr" "aacdemy" "adeginos"
+// let newArr = [testString1, testString2, testString3].sort()
+var testStr1 = testString1.split("").sort()
+var testStr2 = testString2.split("").sort()
+var testStr3 = testString3.split("").sort()
+
+var testStr4 = testStr1.concat(testStr2,testStr3).sort()
+// 
+
+
+const alphabetSoup = (word) =>{
+    return word
+} 
+
+console.log(alphabetSoup(testStr1).join(""))
+console.log(alphabetSoup(testStr2).join(""))
+console.log(alphabetSoup(testStr3).join(""))
 
 
 
 // 4b. Write a function that takes in all three variables and returns a string with all the letters in alphabetical order. Expected output: "aaaacddeeegilmnnorsy"
+// var testStr4 = [testStr1, testStr2, testStr3]
+// var testStr5 = testStr4.sort()
 
+const allVars = (word2) => {
+    return word2
+}
 
+console.log(allVars(testStr4).join(""))
 
```

Comment:
I see what you are doing here, but wrapping the logic in your function is going to make your code much better. Right now the function is taking an argument is just returning it, completely unchanged.